sdks/java/core/build.gradle

```diff
@@ -71,6 +79,13 @@ dependencies {
   shadowTest library.java.slf4j_jdk14
   shadowTest library.java.mockito_core
   shadowTest "com.esotericsoftware.kryo:kryo:2.21"
+  flinkValidatesRunner project(path: project.path, configuration: "shadowTest")
+  flinkValidatesRunner project(path: ":beam-runners-parent:beam-runners-flink_2.11", configuration: "shadow")
+  sparkValidatesRunner project(path: project.path, configuration: "shadowTest")
+  sparkValidatesRunner project(path: ":beam-runners-parent:beam-runners-spark", configuration: "shadow")
+  // TODO: Is there a way to refer to the `provided` configuration from the Spark build?
```

Comment:
yes, "provided" is just another configuration. You should be able to say project(path: ..., configuration: "provided"). Double check the inheritance structure of the "provided" configuration to know what your pulling in.